aten/src/ATen/native/quantized/cpu/conv_serialization.h

```diff
@@ -337,86 +337,38 @@ c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>> deserialize_conv(
   TORCH_INTERNAL_ASSERT(other_flags == 0, "Unexpected flags set in ", flags, ".");
 
   auto& ctx = at::globalContext();
-
-#ifdef USE_FBGEMM
-  if (ctx.qEngine() == at::QEngine::X86) {
-#if AT_MKLDNN_ENABLED()
+  auto eng = ctx.qEngine();
+  if(eng == at::QEngine::X86 || eng == at::QEngine::FBGEMM){
+  #if AT_MKLDNN_ENABLED()
     bool use_onednn = onednn_utils::should_use_onednn_quant(
         weight.value(), transpose, groups, output_padding);
-    if (use_onednn) {
-      return PackedConvWeightsOnednn<kSpatialDim>::prepack(
-        weight.value(),
-        bias,
-        stride,
-        padding,
-        output_padding,
-        dilation,
-        groups,
-        transpose
-      );
-    }
-#endif
-    return PackedConvWeight<kSpatialDim>::prepack(
-      weight.value(),
-      bias,
-      stride,
-      padding,
-      output_padding,
-      dilation,
-      groups,
-      transpose
-    );
-  } // x86
-#endif
-
-#ifdef USE_FBGEMM
-  if (ctx.qEngine() == at::QEngine::FBGEMM) {
-    return PackedConvWeight<kSpatialDim>::prepack(
-      weight.value(),
+    if(use_onednn){
+      return get_device_prepack_fn<kSpatialDim>(at::QEngine::ONEDNN)(weight.value(),
```

Comment:
You removed `#ifdef USE_FBGEMM`. Then, this might be invoked even when `USE_FBGEMM` is not defined. Is it expected?

Reply:
This code is also not meant to be compiled when not using FBGEMM

Reply:
Hi @jgong5  @kimishpatel, Great thanks for the comments . The if-else branch here is intended to handle special engine `x86`, all other engine is handled by general query utils `get_device_preapck_fn`.  

By the way, any query for a unregistered/un-compiled engine prepack function would trigger TOCH_CHECK error.  So, it is within my expectation that these codes is compiled even no FBGEMM, due to the existence of prepack function is always checked.

Original logic in deserialize_conv can be simplified as follows. The x86 is a special engine due to bot onednn/fbgemm has chances to be used.
  ```c++
    #ifdef USE_FBGEMM
      if(engine == at::QEngine::x86){
        use_onednn = se_onednn = onednn_utils::should_use_onednn_quant(....);
        if(use_onednn){
          // call onednn prepack
        }else{
          // call fbgemm prepack
        }
      }
    #endif

    #ifdef USE_FBGEMM
      // handle fbgemm
    #endif

    #ifdef USE_QNNPACK
      // handle qnnpack
    #endif

    #if AT_MKLDNN_ENABLED
      // handle onednn
    #endif

    // fallback path
    // raise error
    TORCH_CHECK("Did not find xxxx")
  ```
  
  I intened to keep the above behavior. So it handles `x86` engine in `if branch`, all other engine is handled through 
  utils `get_device_prepack_fn`. If the prepack function is not compiled for given engine, then an error would be raised.
  ```c++
  if(engine == at::QEngine::x86){
    use_onednn = onednn_utils::should_use_onednn_quant(....);
    if(use_onednn){
      //call onednn prepack
    }else{
      // call fbgemm prepack, if not build with FBGEMM, RAISE ERROR
    }
  }else{
    // use engine to query prepack function, can handle qnnpack, onednn, fbgemm
    // IF NO PREPACK FUNCTION FOUND, RAISE ERROR
    prepack_fn = get_device_prepack_fn(engine)
    prepack_fn(...);
  }
  ```